plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/explorer/BaseExplorerContentProvider.java

```diff
@@ -179,13 +179,37 @@ protected IStatus run(IProgressMonitor monitor) {
 		job.schedule();
 		return stub;
 	}
-
-	protected void refreshViewer(final Object object) {
+	
+	private void asyncViewerRefresh(final Runnable runner) {
```

Comment:
I purposely chose to avoid the word 'update' as it has a connotation with modifying the children of the tree (e.g. update, add, delete, refresh).  I don't really like the name as is but couldnt think of anything better